scripts/training/evaluate_reranker.py

```
@@ -0,0 +1,153 @@
+        by_a.setdefault(r["a_id"], []).append(r)
+    hr = {k: [] for k in k_values}
+    ndcg5 = []
+    for a_id, lst in by_a.items():
+        lst.sort(key=lambda x: x["score"], reverse=True)
+        gold_positions = [i for i, r in enumerate(lst) if r["label"] >= 0.5]
+        if not gold_positions:
+            continue
+        top = max(gold_positions)
+        for k in k_values:
+            hr[k].append(1.0 if min(gold_positions) < k else 0.0)
+        # NDCG@5（单个黄金）
```

Ruff (0.16.3)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)